src/averageAges.js

```diff
@@ -37,7 +48,12 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((person) => person.sex === 'f'
+    && people.some((otherPerson) => otherPerson.mother === person.name))
+    : people.filter((person) => person.sex === 'f');
```

Comment:
let's make a code a little cleaner 
 
```suggestion
  const women = withChildren
    ? people.filter((person) => person.sex === 'f' &&
        people.some((otherPerson) => otherPerson.mother === person.name))
    : people.filter((person) => person.sex === 'f');
```